### DISCUSSION TITLE
What's the best way to integrate Storybook tests with DataDog's dd-trace library?

### DISCUSSION THREAD

**Discussion by Catchpowle** [storybookjs/storybook#28902] [Help]
### Summary

DataDog has a library called [dd-trace](https://github.com/DataDog/dd-trace-js) that can be used to [send test information](https://docs.datadoghq.com/tests/setup/javascript/?tab=githubactions) to DataDog, so you can check the health of your test suite on the CI. It has built-in support for several frameworks but not Storybook Testing yet.

It also has a [manual testing API](https://docs.datadoghq.com/tests/setup/javascript/?tab=githubactions#manual-testing-api), which you can use to instrument non-supported test frameworks. If that's the best route, how would you get the test status data (pass/fail) and the potential error message to pass to Datadog?

**Comment by zeroliu** [comment]
I'm also trying to integrate dd-trace with my new test suite running with storybook vitest plugin. I thought since we now convert storybook stories to vitest tests, dd-trace will automatically pick them up, but they don't seem to work. Only vitest tests are reported to dd-tracer.
